### PR TITLE
Update narze keymaps (planck & ergodox_infinity)

### PR DESCRIPTION
**Pull request by narze**

### CODE REVIEW

File: keyboards/ergodox/keymaps/narze/config.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#ifndef CONFIG_H_
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch? 

```suggestion
#pragma once
```

File: keyboards/ergodox/keymaps/narze/config.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#define CONFIG_H_
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/config.h]:
```suggestion
#define CONFIG_H_
```

File: keyboards/ergodox/keymaps/narze/config.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// #define COMBO_ALLOW_ACTION_KEYS
+
+#define IGNORE_MOD_TAP_INTERRUPT
+#define PERMISSIVE_HOLD
+#define PREVENT_STUCK_MODIFIERS
+
+#undef MOUSEKEY_DELAY
+#define MOUSEKEY_DELAY 100
+
+#endif
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/config.h]:
```suggestion
#endif
```

File: keyboards/ergodox/keymaps/narze/config.h [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#ifndef CONFIG_H_
+#define CONFIG_H_
+
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/config.h]: This line is not actually needed, and should be removed. 

```suggestion
```

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/ergodox/keymaps/narze/config.h [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+#define TAPPING_TERM 150
+
+// Combos not working yet
+// #define COMBO_TERM 20
+// #define COMBO_COUNT 1
+// #define COMBO_ALLOW_ACTION_KEYS
+
+#define IGNORE_MOD_TAP_INTERRUPT
+#define PERMISSIVE_HOLD
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/config.h]: This is not needed anymore, as it's no longer used.  It is now the default behavior. 

```suggestion
// #define STRICT_LAYER_RELEASE
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 6]
```diff
@@ -0,0 +1,852 @@
+#include "ergodox.h"
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: Could you replace this:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 7]
```diff
@@ -0,0 +1,852 @@
...
+// #include "debug.h"
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: Could you remove all of these additional includes?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here. 

Everything but the version.h and keymap_colemak.h.

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 8]
```diff
@@ -0,0 +1,852 @@
...
+#include "version.h"
+#include "eeconfig.h"
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 9]
```diff
@@ -0,0 +1,852 @@
...
+#include "eeprom.h"
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 10]
```diff
@@ -0,0 +1,852 @@
...
+#include "keymap_colemak.h"
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: If you use `SEND_STRING`, and use software colemak (eg, in your OS), then you may want to consider including "sendstring_colemak.h" as well.

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 11]
```diff
@@ -0,0 +1,852 @@
...
+  RAISE,
+  BACKLIT,
+  EXT_PLV,
+  SDTOGG, // Toggle SuperDuper
+  EPRM,
+  VRSN,
+  RGB_SLD
+};
+
+enum functions {
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: Could you actually remove this? 

We have deprecated macros/functions in favor of `process_record_user`.

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 12]
```diff
@@ -0,0 +1,852 @@
...
+  M_SFT_PC, // SFT_T(KC_RPRN)
+};
+
+// Timer for custom mod tap
+static uint16_t m_gui_unds_timer;
+static uint16_t m_sft_po_timer;
+static uint16_t m_sft_pc_timer;
+
+// Fillers to make layering more clear
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 13]
```diff
@@ -0,0 +1,852 @@
...
+#define XXXXXXX KC_NO
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 14]
```diff
@@ -0,0 +1,852 @@
...
+ *   `----------------------------------'                                       `----------------------------------'
+ *                                        ,-------------.       ,-------------.
+ *                                        | Enter| LGui |       | Alt  |Ctrl/Esc|
+ *                                 ,------|------|------|       |------+--------+------.
+ *                                 |      |      | Home |       | PgUp |        |      |
+ *                                 | Space| Back |------|       |------| Back   |Enter |
+ *                                 |      | Space| End  |       | PgDn | Space  |      |
+ *                                 `--------------------'       `----------------------'
+ */
+[_QWERTY] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: We're phasing out `KEYMAP` in favor of `LAYOUT`.  If you could change this? 

```suggestion
[_QWERTY] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 15]
```diff
@@ -0,0 +1,852 @@
...
+ *   `----------------------------------'                                       `----------------------------------'
+ *                                        ,-------------.       ,-------------.
+ *                                        | Enter| LGui |       | Alt  |Ctrl/Esc|
+ *                                 ,------|------|------|       |------+--------+------.
+ *                                 |      |      | Home |       | PgUp |        |      |
+ *                                 | Space| Back |------|       |------| Back   |Enter |
+ *                                 |      | Space| End  |       | PgDn | Space  |      |
+ *                                 `--------------------'       `----------------------'
+ */
+[_COLEMAK] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: We're phasing out `KEYMAP` in favor of `LAYOUT`.  If you could change this? 

```suggestion
[_COLEMAK] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 16]
```diff
@@ -0,0 +1,852 @@
...
+             LT(_SYMB, KC_NO),   KC_J,   KC_L,  KC_U,   KC_Y,   KC_SCLN,       KC_BSLS,
+                          KC_H,   KC_N,  KC_E,   KC_I,   KC_O,          KC_QUOT,
+             MEH_T(KC_NO),KC_K,   KC_M,  KC_COMM,KC_DOT, LT(_SUPERDUPER, KC_SLSH),   SFT_PC,
+                                  RAISE, GUI_MINS,KC_RALT,KC_RCTL, LT(_LOWER, KC_RBRC),
+             KC_LALT,        CTL_T(KC_ESC),
+             KC_PGUP,
+             KC_PGDN,KC_BSPC, KC_ENT
+    ),
+/* Qwerty on software Colemak : Useful for gaming with qwerty keymaps! */
+[_QWOC] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: We're phasing out `KEYMAP` in favor of `LAYOUT`.  If you could change this? 

```suggestion
[_QWOC] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 17]
```diff
@@ -0,0 +1,852 @@
...
+ *   `----------------------------------'                                       `----------------------------------'
+ *                                        ,-------------.       ,-------------.
+ *                                        |      |      |       |      |      |
+ *                                 ,------|------|------|       |------+------+------.
+ *                                 |      |      |      |       |      |      |      |
+ *                                 |      |      |------|       |------|      |      |
+ *                                 |      |      |      |       |      |      |      |
+ *                                 `--------------------'       `--------------------'
+ */
+[_LOWER] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: We're phasing out `KEYMAP` in favor of `LAYOUT`.  If you could change this? 

```suggestion
[_LOWER] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 18]
```diff
@@ -0,0 +1,852 @@
...
+ *   `----------------------------------'                                       `----------------------------------'
+ *                                        ,-------------.       ,-------------.
+ *                                        |      |      |       |      |      |
+ *                                 ,------|------|------|       |------+------+------.
+ *                                 |      |      |      |       |      |      |      |
+ *                                 |      |      |------|       |------|      |      |
+ *                                 |      |      |      |       |      |      |      |
+ *                                 `--------------------'       `--------------------'
+ */
+[_RAISE] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: We're phasing out `KEYMAP` in favor of `LAYOUT`.  If you could change this? 

```suggestion
[_RAISE] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 19]
```diff
@@ -0,0 +1,852 @@
...
+ *   `----------------------------------'                                       `----------------------------------'
+ *                                        ,-------------.       ,-------------.
+ *                                        | Enter| LGui |       | Alt  |Ctrl/Esc|
+ *                                 ,------|------|------|       |------+--------+------.
+ *                                 |      |      | Home |       | PgUp |        |      |
+ *                                 | Space| Back |------|       |------| Back   |Enter |
+ *                                 |      | Space| End  |       | PgDn | Space  |      |
+ *                                 `--------------------'       `----------------------'
+ */
+[_PLOVER] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
[_PLOVER] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 20]
```diff
@@ -0,0 +1,852 @@
...
+ *   `----------------------------------'                                       `----------------------------------'
+ *                                        ,-------------.       ,-------------.
+ *                                        |      |      |       |      |        |
+ *                                 ,------|------|------|       |------+--------+------.
+ *                                 |      |      |      |       |      |        |      |
+ *                                 | Shift|      |------|       |------|        |Shift |
+ *                                 |      |      |      |       |      |        |      |
+ *                                 `--------------------'       `----------------------'
+ */
+[_SUPERDUPER] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: We're phasing out `KEYMAP` in favor of `LAYOUT`.  If you could change this? 


```suggestion
[_SUPERDUPER] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 21]
```diff
@@ -0,0 +1,852 @@
...
+             _______, _______,  _______,   S(LGUI(KC_LBRC)), S(LGUI(KC_RBRC)),   _______,   _______,
+                      KC_LEFT,  KC_DOWN,   KC_UP,    KC_RGHT,   KC_DEL,    _______,
+             _______, _______,  _______,   _______,  _______,   _______,   _______,
+                                _______,   _______,  _______,   _______,   _______,
+             _______,        _______,
+             _______,
+             _______,_______, KC_LSFT
+    ),
+// Intermediate keymaps for SuperDuper (Combo keys does not work on Infinity yet)
+[_SUPER] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
[_SUPER] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 22]
```diff
@@ -0,0 +1,852 @@
...
+             _______, _______,  _______,   _______,  _______,   _______,   _______,
+             _______, _______,  _______,   _______,  _______,   _______,   _______,
+                      _______,  _______,   _______,  _______,   _______,    _______,
+             _______, _______,  _______,   _______,  _______,   _______,   _______,
+                                _______,   _______,  _______,   _______,   _______,
+             _______,        _______,
+             _______,
+             _______,_______, _______
+    ),
+[_DUPER] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
[_DUPER] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 23]
```diff
@@ -0,0 +1,852 @@
...
+ *   `----------------------------------'                                       `----------------------------------'
+ *                                        ,-------------.       ,-------------.
+ *                                        |      |      |       |      |        |
+ *                                 ,------|------|------|       |------+--------+------.
+ *                                 |      |      |      |       |      |        |      |
+ *                                 | Left |      |------|       |------|        | Left |
+ *                                 | Click|      |      |       |      |        | Click|
+ *                                 `--------------------'       `----------------------'
+ */
+[_MOUSE] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
[_MOUSE] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 24]
```diff
@@ -0,0 +1,852 @@
...
+ *   `----------------------------------'                                       `----------------------------------'
+ *                                        ,-------------.       ,-------------.
+ *                                        |      |      |       |      |        |
+ *                                 ,------|------|------|       |------+--------+------.
+ *                                 |      |      |      |       |      |        |      |
+ *                                 |      |      |------|       |------|        |      |
+ *                                 |      |      |      |       |      |        |      |
+ *                                 `--------------------'       `----------------------'
+ */
+[_ADJUST] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
[_ADJUST] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 25]
```diff
@@ -0,0 +1,852 @@
...
+ *                                        ,-------------.       ,-------------.
+ *                                        |Animat|      |       |Toggle|Solid |
+ *                                 ,------|------|------|       |------+------+------.
+ *                                 |Bright|Bright|      |       |      |Hue-  |Hue+  |
+ *                                 |ness- |ness+ |------|       |------|      |      |
+ *                                 |      |      |      |       |      |      |      |
+ *                                 `--------------------'       `--------------------'
+ */
+// SYMBOLS
+[_SYMB] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
[_SYMB] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 26]
```diff
@@ -0,0 +1,852 @@
...
+ *                                        ,-------------.       ,-------------.
+ *                                        |      |      |       |      |      |
+ *                                 ,------|------|------|       |------+------+------.
+ *                                 |      |      |      |       |      |      |Brwser|
+ *                                 |      |      |------|       |------|      |Back  |
+ *                                 |      |      |      |       |      |      |      |
+ *                                 `--------------------'       `--------------------'
+ */
+// MEDIA AND MOUSE
+[_MDIA] = KEYMAP(
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
[_MDIA] = LAYOUT_ergodox(
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 27]
```diff
@@ -0,0 +1,852 @@
...
+                 _______, _______, _______, _______, _______, KC_MPLY,
+       _______,  _______, _______, KC_MPRV, KC_MNXT, _______, _______,
+                          KC_VOLU, KC_VOLD, KC_MUTE, _______, _______,
+       _______, _______,
+       _______,
+       _______, _______, KC_WBAK
+)
+};
+
+void persistant_default_layer_set(uint16_t default_layer) {
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: Could you remove this function?  

It's been depreciated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 28]
```diff
@@ -0,0 +1,852 @@
...
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        persistant_default_layer_set(1UL<<_QWERTY);
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
        set_single_persistent_default_layer(_QWERTY);
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 29]
```diff
@@ -0,0 +1,852 @@
...
+
+        // key_combos[CB_SUPERDUPER].keys = superduper_combos[_QWERTY];
+        // eeprom_update_byte(EECONFIG_SUPERDUPER_INDEX, _QWERTY);
+      }
+      return false;
+      break;
+    case COLEMAK:
+      if (record->event.pressed) {
+        persistant_default_layer_set(1UL<<_COLEMAK);
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
        set_single_persistent_default_layer(_COLEMAK);
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 30]
```diff
@@ -0,0 +1,852 @@
...
+
+        // key_combos[CB_SUPERDUPER].keys = superduper_combos[_COLEMAK];
+        // eeprom_update_byte(EECONFIG_SUPERDUPER_INDEX, _COLEMAK);
+      }
+      return false;
+      break;
+    case QWOC:
+      if (record->event.pressed) {
+        persistant_default_layer_set(1UL<<_QWOC);
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]:
```suggestion
        set_single_persistent_default_layer(_QWOC);
```

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 31]
```diff
@@ -0,0 +1,852 @@
...
+
+        // if (superduper_enabled) {
+        //   set_superduper_key_combos();
+        // } else {
+        //   clear_superduper_key_combos();
+        // }
+      }
+      return false;
+      break;
+    case EPRM:
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: This is no longer needed, and is a core keycode. 

https://docs.qmk.fm/#/quantum_keycodes?id=qmk-keycodes
`EEP_RST`

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 32]
```diff
@@ -0,0 +1,852 @@
...
+//   } else {
+//     layer_off(_SUPERDUPER);
+//     ergodox_board_led_off();
+//     unregister_mods(MOD_BIT(KC_LGUI) | MOD_BIT(KC_LCTL) | MOD_BIT(KC_LALT)); // Sometimes mods are held, unregister them
+//   }
+// }
+
+// Macros
+
+const uint16_t PROGMEM fn_actions[] = {
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: Could you remove the `fn_actions` block?  All of it's features have been added as keycodes, and been deprecated in favor of `process_record_user`.

File: keyboards/ergodox/keymaps/narze/keymap.c [hunk 33]
```diff
@@ -0,0 +1,852 @@
...
+  [M_GUI_UNDS] = ACTION_MACRO_TAP(M_GUI_UNDS),
+  [M_SFT_PO] = ACTION_MACRO_TAP(M_SFT_PO),
+  [M_SFT_PC] = ACTION_MACRO_TAP(M_SFT_PC),
+};
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
```
Comment (drashna) [inline keyboards/ergodox/keymaps/narze/keymap.c]: Could you remove the `action_get_macro` function?  It's been deprecated in favor of `process_record_user`.

**Comment by narze** [comment]
I haven't update the repo over than a year 😂
I'll resolve all listed issues and will resubmit this pr 👍

**Comment by drashna** [comment]
Well, a lot has changed! 

If you need any help with that: https://docs.qmk.fm/#/newbs_best_practices

**Comment by drashna** [comment]
Any update on this?

**Comment by narze** [comment]
I think I should open a new pr.

**Comment by narze** [comment]
Close in favor of #6652

**Pull request closed without merging**